src/BrewingRecipe.cpp

```diff
@@ -0,0 +1,324 @@
+
+#include "Globals.h"  // NOTE: MSVC stupidness requires this to be the same across all modules
+
+#include "BrewingRecipe.h"
+#include "Item.h"
+
+#include <fstream>
+
+#define BREWING_RECIPE_FILE "brewing.txt"
+
+
+
+
+
+typedef std::list<cBrewingRecipe::cRecipe> RecipeList;
+
+
+
+
+
+struct cBrewingRecipe::sBrewingRecipeState
+{
+	RecipeList Recipes;
+};
+
+
+
+
+
+cBrewingRecipe::cBrewingRecipe()
+	: m_pState(new sBrewingRecipeState)
+{
+	ReloadRecipes();
+}
+
+
+
+
+
+cBrewingRecipe::~cBrewingRecipe()
+{
+	ClearRecipes();
+	delete m_pState;
+	m_pState = nullptr;
+}
+
+
+
+
+
+void cBrewingRecipe::ReloadRecipes(void)
+{
+	ClearRecipes();
+	LOGD("Loading brewing recipes...");
+
+	std::ifstream f(BREWING_RECIPE_FILE, std::ios::in);
```

Comment:
Nah, using streams for file IO is okay. I had some reservations for using them to convert between strings and numbers.